### PR TITLE
fix(rome_js_formatter): print empty statements as empty blocks

### LINKED ISSUE #2448: 🐛 loop/if with empty statement body
### Environment information

```bash
WSL (W11/Ubuntu)
Yarn 3.2.0
```

### What happened?

1. Create project using yarn 3.2.0
2. Install Rome 
3. Run `yarn format` (`rome format .`)
4. Rerun `yarn`
```
project$ yarn format
Formatted x files in xms

project$ yarn
project/.yarn/releases/yarn-3.2.0.cjs:37184
                                                !e && !this.buffer.length && !this[sA] && this.emit("drain");
                                                ^

SyntaxError: Unexpected token '!'
    at Object.compileFunction (node:vm:352:18)
    at wrapSafe (node:internal/modules/cjs/loader:1031:15)
    at Module._compile (node:internal/modules/cjs/loader:1065:27)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:79:12)
    at node:internal/main/run_main_module:17:47

```

### Expected result

Rome should correctly format cjs without breaking it or not touch cjs at all.

### Code of Conduct

- [X] I agree to follow Rome's Code of Conduct

### PR DESCRIPTION
**Pull request by ematipico**
## Summary

Fixes #2448

## Test Plan

Added new cases, one for the `if` and one for `do while`

### CODE REVIEW

File: crates/rome_js_formatter/src/js/statements/empty_statement.rs [hunk 1]
```diff
@@ -1,13 +1,22 @@
...
 use rome_js_syntax::JsEmptyStatementFields;
+use rome_js_syntax::{JsEmptyStatement, JsSyntaxKind};
+use rome_rowan::AstNode;
 
 impl FormatNode for JsEmptyStatement {
     fn format_fields(&self, formatter: &Formatter) -> FormatResult<FormatElement> {
         let JsEmptyStatementFields { semicolon_token } = self.as_fields();
-
-        Ok(formatter.format_replaced(&semicolon_token?, empty_element()))
+        let parent_kind = self.syntax().parent().map(|p| p.kind());
```
Comment (NicholasLYang) [inline crates/rome_js_formatter/src/js/statements/empty_statement.rs:10]: Nit: this pattern shows up a couple times. Should we have a function that gets the parent kind directly?

**Review by NicholasLYang — APPROVED** [review]

**Comment by ematipico** [comment]
> When it comes to unifying the printing behavior of empty statements, I would suggest we go the opposite way and extend the behavior of `if` statements (always insert a block) to all other affected statements (`for` loops, `for in` loops, `for of` loops, `while` loops, `do while` loops and `with` blocks). The reasoning behind this is that Rome JS had a `useBlockStatements` lint rule that emitted a fixable warning for these nodes if they were using a single statement as their body, but for this kind of "style fix" we might as well rewrite it directly in the formatter.

Does that mean that we want to intentionally diverge from prettier?

**Comment by leops** [comment]
> Does that mean that we want to intentionally diverge from prettier?

Well at least I think we should discuss whether we want to have code-style related lint rules in the analyzer or to enforce these decisions in the formatter directly. Ultimately the decision is probably about how disruptive the change is (how often are these single-statement syntax actually used) vs. how confusing it is for users to see fixable diagnostics in their code for things they'd expect the formatter to take care of.

**Comment by ematipico** [comment]
> > Does that mean that we want to intentionally diverge from prettier?
> 
> Well at least I think we should discuss whether we want to have code-style related lint rules in the analyzer or to enforce these decisions in the formatter directly. Ultimately the decision is probably about how disruptive the change is (how often are these single-statement syntax actually used) vs. how confusing it is for users to see fixable diagnostics in their code for things they'd expect the formatter to take care of.

I completely forgot about that lint rule. Yes, in Rome classic there was this idea where "coding style rules" should not exists because we own the formatter too. 

This is the rule FYI: https://github.com/rome/tools/blob/archived-js/internal/compiler/lint/rules/js/useBlockStatements.test.md

At this point I think we should just diverge from prettier. I agree with you @leops

**Comment by yassere** [comment]
> At this point I think we should just diverge from prettier. I agree with you @leops

I don't think we should. This is a style decision, and we should have a very high bar for where we diverge from Prettier purely for style reasons (rather than implementation complexity).

We don't necessarily need to add every lint rule that existed in Rome classic. But if we do end up with some "code style rules", I think that's ok because we've intentionally relinquished some of our control over the formatting output by targeting Prettier equivalence.

**Comment by ematipico** [comment]
@yassere Could let us know why though? Why you think we should not put blocks and why we don't need that lint rule anymore.

I think it makes sense to have blocks. Having empty statements can lead to errors, same for expressions without blocks.

**Comment by NicholasLYang** [comment]
Personally I view diverging from Prettier as a large enough reason. If we get far enough in linting that we create this rule again, we can consider diverging then. But until that time, we should do what's best for the formatter, and not what might be best for the linter. IMO, as long as we're not seriously inhibiting future tools, we should do what's best for the tools that currently exist. Each of our tools needs to gain adoption and the promise of a linter feature does not outweigh breaking prettier compatibility, at least for me.

**Review by yassere — COMMENTED** [review]
@ematipico I think it makes sense to have blocks too. Maybe that's something we can have as a safe autofix for the linter. But for the _formatter_, I think we need to respect our commitment to match Prettier's styling decisions as much as possible, unless we have a very good reason not to.

Given our previous messaging, intentionally deviating from Prettier's style is not a decision we should make lightly. Each case should be discussed in its own issue so that we can make a strong case to justify that divergence and document it.

**Pull request closed without merging**